doc/modules/lmnn.rst

```diff
@@ -0,0 +1,161 @@
+dimensional space, LMNN constructs a set of hyper-spheres centered at each
+training point that are supposed to contain only samples that share the same
+label as the central point.
+
+In the beginning of the algorithm each training sample fixes :math:`k`
+target neighbors, namely the :math:`k` nearest training samples that share
+the same label according to some initial metric - most commonly the Euclidean.
+
+The first goal of the algorithm is to find a metric such that the distances of
+each sample to its target neighbors are minimized. The second goal of the
+algorithm is to create a large margin between the farthest target neighbor
+of each sample and data samples that belong to different classes, thereby
+enclosing the target neighbors in a sphere around the reference sample. Data
+samples from different classes that violate this margin are called impostors.
+
+Combined with a nearest neighbors classifier (:class:`classification.KNeighborsClassifier`)
+this method is attractive for classification because it requires no
+modification or extension for multi-class (as opposed to binary) problems
+and only a single parameter (``n_neighbors``) has to be selected by the user
+before training.
+
+Large Margin Nearest Neighbor classification has been proven to work well in
+practice for data sets of varying size and difficulty.
+
+
+.. Enter ../auto_examples/classification/images/sphx_glr_plot_lda_lmnn_001.png
+        :target: ../auto_examples/classification/plot_lda_lmnn.html
+        :scale: 80
+
+.. Enter centered:: |lmnn|
+
+.. The plot shows decision boundaries for nearest neighbor classification and
+    large margin nearest neighbor classification. The bottom row shows the
+    same comparison after first doing PCA on the inputs.
+
+.. Enter topic:: Examples:
+    .. Comment
+    :ref:`sphx_glr_auto_examples_classification_plot_lda_lmnn.py`: Comparison
+    of LDA and LMNN on synthetic data.
+
+Dimensionality reduction
+========================
+
+:class:`lmnn.LargeMarginNearestNeighbor` can be used to
+perform supervised dimensionality reduction. The input data are projected
+onto a linear subspace consisting of the directions which maximize the nearest
+neighbor classification accuracy. Therefore the separation between clusters
+of samples of the same class is also maximized.
+
+This is implemented in :func:`lmnn.LargeMarginNearestNeighbor.transform`.
+The desired dimensionality can be set using the ``n_features_out`` parameter
+in the constructor.
+
+.. Comment topic:: Examples:
+    ..
+    :ref:`sphx_glr_auto_examples_decomposition_plot_pca_vs_lda_vs_lmnn.py`:
+    Comparison of PCA, LDA and LMNN
+    for dimensionality reduction of the Olivetti faces dataset
+
+Mathematical formulation
+========================
+
+The LMNN objective function consists of two competing terms, the pull loss
+that pulls target neighbors closer to their reference samples and the push
+loss that pushes impostors away:
+
+.. math::
```

Comment:
it would be nice to have the minimization problem written down instead of just the objective function?